overlord/servicestate/quotas.go

```diff
@@ -60,27 +60,29 @@ func GetQuota(st *state.State, name string) (*quota.Group, error) {
 	return allGrps[name], nil
 }
 
-// UpdateQuotas will update the state quota group map with the provided quota
-// groups. The groups provided will replace group states if present or be added
-// on top of the current set of quota groups in the state, and verified for
-// consistency before committed to state. When adding sub-groups, both the
-// parent and the sub-group must be added at once since the sub-group needs to
-// reference the parent group and vice versa to be fully consistent.
-func UpdateQuotas(st *state.State, grps ...*quota.Group) error {
+// PatchQuotasState will update the state quota group map with the provided quota
+// groups. It returns the full set of all quota groups after a successful
+// update for convenience. The groups provided will replace group states if
+// present or be added on top of the current set of quota groups in the state,
+// and verified for consistency before committed to state. When adding
+// sub-groups, both the parent and the sub-group must be added at once since the
+// sub-group needs to reference the parent group and vice versa to be fully
+// consistent.
+func PatchQuotasState(st *state.State, grps ...*quota.Group) (map[string]*quota.Group, error) {
```

Comment:
In the end probably not, I was using it from tests since there wasn't yet a higher level method to create a group in state but arguably the tests could use mgr.CreateQuota instead eventually

Reply:
Happy to apply the attached patch if we want to unexport this or do it in a followup.

[unexport-quota-state.txt](https://github.com/snapcore/snapd/files/6398232/unexport-quota-state.txt)

Reply:
Once that is applied (if desired) I can rename to ` patchQuotas`

Reply:
that patch is awkward because the methods on the manager should really just be functions taking state

Reply:
I did not apply this, I think it's easier to do as a followup

Reply:
Yeah, followup sounds ok to me.

Reply:
yes, follow-up is good,  both unexporting and more conventional name would be good

Reply:
done in https://github.com/snapcore/snapd/pull/10216